### PR TITLE
perf(headers): a single set-cookie

### PR DESCRIPTION
**Pull request by tsctx**
**benchmarks/fetch/headers-length32.mjs**
- main
```
benchmark              time (avg)             (min … max)       p75       p99      p999
--------------------------------------------------------- -----------------------------
Headers@@iterator   5'731 ns/iter     (4'300 ns … 516 µs)  5'300 ns 11'600 ns    115 µs
```

- this patch
```
benchmark              time (avg)             (min … max)       p75       p99      p999
--------------------------------------------------------- -----------------------------
Headers@@iterator   5'151 ns/iter     (4'100 ns … 689 µs)  4'800 ns 11'100 ns    109 µs
```

### CODE REVIEW

File: benchmarks/fetch/headers-length32.mjs [hunk 1]
```diff
@@ -19,7 +19,7 @@ const headers = new Headers(
     'Width',
     'Accept-CH',
     'Via',
-    'Refresh',
```
Comment (mcollina) [inline benchmarks/fetch/headers-length32.mjs:22]: why was this removed?

Comment (tsctx) [reply benchmarks/fetch/headers-length32.mjs:22]: This is because I replaced it with set-cookie.
The length, which is a fast-path condition, must be 32.

**Review by KhafraDev — APPROVED** [review]

**Review by Uzlopak — APPROVED** [review]
I added a unit test to ensure it works as expected.

**Review by mcollina — APPROVED** [review]
lgtm
